### ISSUE TITLE
Segmentation fault during Qlora fine tuning on ARC 770

### ISSUE THREAD

**Issue by tsantra** [intel-analytics/BigDL#9412]
Model: llama-2-7b-hf
Ubuntu :22.04

xpu-smi discovery:
<img width="613" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/a8b35b15-b758-42f0-b2b6-82befc064161">

uname -r 
<img width="119" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/65ef1c09-4082-4f34-b961-ce24538e328c">

Steps followed:
1. Created conda env (followed instructions in Repo example)
2. Then initialized oneAPI

<img width="471" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/f704058d-7232-4ef4-8747-5d8758099f14">

3. Used default dataset for model finetuning:

<img width="738" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/603efd90-ffee-407f-9454-d7015b9fbbc7">

4. sycl-ls info:
<img width="763" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/8ee32033-3af6-4038-a381-8600a6df915b">

5. lscpu

<img width="787" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/cb149fc7-b667-4c6b-ab60-580f157cb35a">

**Comment by rnwang04** [comment]
Yeah, I have reproduced this error. I found on your machine, inference works fine, but once finetuing, it will seg fault at rope.
Still no idea about the root cause.
By the way, below is our machine's linux version and driver version:
![image](https://github.com/intel-analytics/BigDL/assets/105281011/ee7e5955-3d09-4dfb-8ea1-c3f6de43172d)

Not sure whether this error is caused by above version mismatch.

@qiuxin2012 @yangw1234  any suggestions ?

**Comment by tsantra** [comment]
@rnwang04   when I do apt list -a linux-generic-hwe-22.04, I don't see 5.19 kernel

<img width="708" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/0a164b94-e913-4633-9ec2-023f836540fd">

If kernel mismatch is the issue, how do I install 5.19? The https://dgpu-docs.intel.com/driver/client/overview.html says 5.19 should be there with linux-generic-hwe-22.04, but then its not as shown in the screen shot above.

**Comment by tsantra** [comment]
any suggestion on what to do next? Thank you!

**Comment by rnwang04** [comment]
We have verified locally that `6.2.0-35-generic + Intel(R) Arc(TM) A770 Graphics 1.3 [1.3.26690] ` can run QLoRA on our local A770 machine. So it seems not version issue and I have checked the python package version is all the same.
But I found that on your mechine, the output of `sudo xpu-smi stats -d 0`'s output is a little strange:
![image](https://github.com/intel-analytics/BigDL/assets/105281011/de01a2e6-5e52-47dc-8a8b-469a5f32a809)

On our,  it is:
![image](https://github.com/intel-analytics/BigDL/assets/105281011/4fc3df32-9ca3-4130-9316-cd4ee22cd387)

Not sure whether this issue is caused by some installation error or anything else.

**Comment by rnwang04** [comment]
Below steps are how we setup our arc env on ubuntu 22.04.3, maybe you can refer to this.
```bash
Commands on ubuntu 22.04.3:
```bash
# install arc driver 
sudo apt-get install -y gpg-agent wget
wget -qO - https://repositories.intel.com/graphics/intel-graphics.key | \
  sudo gpg --dearmor --output /usr/share/keyrings/intel-graphics.gpg
echo 'deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/graphics/ubuntu jammy arc' | \
  sudo tee  /etc/apt/sources.list.d/intel.gpu.jammy.list


# downgrade kernel
  
sudo apt-get update && sudo apt-get install  -y --install-suggests  linux-image-5.19.0-41-generic

sudo sed -i "s/GRUB_DEFAULT=.*/GRUB_DEFAULT=\"1> $(echo $(($(awk -F\' '/menuentry / {print $2}' /boot/grub/grub.cfg \
| grep -no '5.19.0-41' | sed 's/:/\n/g' | head -n 1)-2)))\"/" /etc/default/grub

sudo  update-grub

sudo reboot
# As 5.19's kernel doesn't has any arc graphic driver. The machine may not start the desktop correctly, but we can use the ssh to login. 
# Or you can select 5.19's recovery mode in the grub, then choose resume to resume the normal boot directly.

# remove latest kernel

sudo apt purge linux-image-6.2.0-*

sudo apt autoremove

sudo reboot

# install drivers

sudo apt-get update

sudo apt-get -y install \
    gawk \
    dkms \
    linux-headers-$(uname -r) \
    libc6-dev
	
sudo apt-get install -y intel-platform-vsec-dkms intel-platform-cse-dkms intel-i915-dkms intel-fw-gpu

sudo apt-get install -y gawk libc6-dev udev\
  intel-opencl-icd intel-level-zero-gpu level-zero \
  intel-media-va-driver-non-free libmfx1 libmfxgen1 libvpl2 \
  libegl-mesa0 libegl1-mesa libegl1-mesa-dev libgbm1 libgl1-mesa-dev libgl1-mesa-dri \
  libglapi-mesa libgles2-mesa-dev libglx-mesa0 libigdgmm12 libxatracker2 mesa-va-drivers \
  mesa-vdpau-drivers mesa-vulkan-drivers va-driver-all vainfo
  
sudo reboot

# Configuring permissions

sudo gpasswd -a ${USER} render
newgrp render

# Verify the device is working with i915 driver
sudo apt-get install -y hwinfo
hwinfo --display


# install one api

wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor | sudo tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null
echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
sudo apt update
sudo apt install intel-basekit
```
```
